docs/blog/announcing-vite3.md

```diff
@@ -0,0 +1,228 @@
+
+### Vite CLI
+
+<pre style="background-color: var(--vp-code-block-bg);padding:2em;border-radius:8px;max-width: 100%">
+  <span style="color:lightgreen"><b>VITE</b></span> <span style="color:lightgreen">v3.0.0</span>  <span style="color:gray">ready in <b>320</b> ms</span>
+
+  <span style="color:lightgreen"><b>➜</b></span>  <span style="color:white"><b>Local</b>:</span>   <span style="color:cyan">http://127.0.0.1:5173/</span>
+  <span style="color:green"><b>➜</b></span>  <span style="color:gray"><b>Network</b>: use --host to expose</span>
+</pre>
+
+Apart from the CLI’s aesthetics improvements, you’ll notice that the default dev server port is now 5173 and the preview server listening at 4173. This change ensures Vite will avoid collisions with other tools.
+
+### Improved WebSocket Connection Strategy
+
+One of the pain points of Vite v2 was configuring the server when running behind a proxy. Vite v3 changes the default connection scheme so it works out of the box in most scenarios. All these setups are now tested as part of the Vite Ecosystem CI through [`vite-setup-catalogue`](https://github.com/sapphi-red/vite-setup-catalogue).
+
+### ESM SSR build by default
+
+Most SSR frameworks in the ecosystem were already using ESM builds. So, Vite v3 makes ESM the default format for SSR builds. This allows us to streamline previous [SSR externalization heuristics](https://vitejs.dev/guide/ssr.html#ssr-externals), externalizing dependencies by default.
+
+### Cold Start Improvements
+
+Vite now avoids full reload during cold start when imports are injected by plugins while crawling the initial statically imported modules ([#8869](https://github.com/vitejs/vite/issues/8869)).
+
+In Vite 2.9, both the scanner and optimizer were run in the background. In the best scenario, where the scanner would find every dependency, no reload was needed in cold start. But if the scanner missed a dependency, a new optimization phase and then a reload were needed. Vite was able to avoid some of these reloads in v2.9, as we detected if the new optimized chunks were compatible with the ones the browser had. But if there was a common dep, the sub-chunks could change and a reload was required to avoid duplicated state. In Vite 3, the optimized deps aren't handed to the browser until the crawling of static imports is done. A quick optimization phase is issued if there is a missing dep (for example, injected by a plugin), and only then, the bundled deps are sent. So, a page reload is no longer needed for these cases.
```

Comment:
This two paragraph sounds too internal for a public announcement 🤔 Maybe we could simplify it as something like:

```
Vite now avoids a full reload during cold start by eagerly analyzing missing dependencies in requested files. In contrast to Vite 2.9, where requested files are sent as-is.
```

and have a details-summary of an in-depth explanation.

Reply:
But the graphs look good for the announcement? I'm afraid that without the explanation people will not understand them.

Reply:
I feel like the explanation needs a lot of context to understand, and with a large amount of text, it'll most likely be glanced over.

Maybe we could simplify the explanation a bit, and/or split it into multiple paragraphs. But I think the best case is if we wrote an in-depth guide in the docs and link to it in the future.

Reply:
If I simplify the paragraph, and we leave the image more just to show that it takes less time in 3, is that ok? Or do you prefer we remove the diagrams too?

Reply:
Pushed this change, we can still refine it if you prefer that.

Reply:
I think it's fine to show the diagram to entice them 😄

Reply:
Looks like you moved the diagram into `details`, but I'm fine if we leave it out too.

Reply:
Oh, ok, I'll move it out, I like the diagram 👀